src/server/exemption/site-details/coordinate-system/controller.js

```diff
@@ -67,10 +75,14 @@ export const coordinateSystemSubmitController = {
 
         const { projectName } = getExemptionCache(request)
 
+        const site = setSiteData(request)
+        const { queryParams } = site
+
         if (!err.details) {
           return h
             .view(COORDINATE_SYSTEM_VIEW_ROUTE, {
               ...coordinateSystemSettings,
+              backLink: coordinateSystemSettings.backLink + queryParams,
```

Comment:
Just wondering, if the current exemption can be retrieved from cache and isn't in the URL, couldn't the same be done for the current site? Then no need to pass querystrings around.